src/boards/include/boards/raspberry_pi_pi500_rp2040.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2024 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+// -----------------------------------------------------
+// NOTE: THIS HEADER IS ALSO INCLUDED BY ASSEMBLER SO
+//       SHOULD ONLY CONSIST OF PREPROCESSOR DIRECTIVES
+// -----------------------------------------------------
+
+#ifndef _BOARDS_RASPBERRY_PI_PI500_RP2040_H
+#define _BOARDS_RASPBERRY_PI_PI500_RP2040_H
+
+pico_board_cmake_set(PICO_PLATFORM, rp2040)
+
+// For board detection
+#define RASPBERRY_PI_PI500_RP2040
+
+// --- UART ---
+#ifndef PICO_DEFAULT_UART
+#define PICO_DEFAULT_UART 0
+#endif
+#ifndef PICO_DEFAULT_UART_TX_PIN
+#define PICO_DEFAULT_UART_TX_PIN 16  // Pi 500 debug UART on GP16
+#endif
+#ifndef PICO_DEFAULT_UART_RX_PIN
+#define PICO_DEFAULT_UART_RX_PIN 1   // Standard RX pin
+#endif
```

Comment:
According to your comments below, GP1 is used as a matrix row, so you shouldn't define it here.